src/test/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacadeTest.java

```diff
@@ -44,66 +49,237 @@ public void setUp() {
+        assertNotNull(createBookingResponse.getEntity());
+        if (createBookingResponse.getEntity() instanceof EventBookingDTO) {
             eventBookingDTO = (EventBookingDTO) createBookingResponse.getEntity();
             // Check that the returned entity is an EventBookingDTO and the ID of the user who created the booking matches
             assertEquals(studentLogin.user.getId(), eventBookingDTO.getUserBooked().getId());
+        } else {
+            fail("The returned entity is not an instance of EventBookingDTO.");
         }
+        // This is probably redundant at this point, but we use this object later so might as well.
         assertNotNull(eventBookingDTO);
 
-        // --- Check whether we are leaking PII to event managers
-        HttpServletRequest getEventBookingsByIdRequest = createNiceMock(HttpServletRequest.class);
-        expect(getEventBookingsByIdRequest.getCookies()).andReturn(new Cookie[] { eventManagerLogin.cookie }).atLeastOnce();
+        // --- Check whether what we get as event managers contains the right amount of information
+        HttpServletRequest getEventBookingsByIdRequest = createRequestWithCookies(new Cookie[] { eventManagerLogin.cookie });
         replay(getEventBookingsByIdRequest);
 
         Response getEventBookingsByIdResponse = eventsFacade.getEventBookingsById(getEventBookingsByIdRequest, eventBookingDTO.getBookingId().toString());
         assertNotNull(getEventBookingsByIdResponse.getEntity());
-        if (getEventBookingsByIdResponse.getEntity() instanceof EventBookingDTO) {
-            assertNotNull(((EventBookingDTO) getEventBookingsByIdResponse.getEntity()).getUserBooked());
-            assertEquals(((EventBookingDTO) getEventBookingsByIdResponse.getEntity()).getUserBooked().getClass(), UserSummaryWithEmailAddressDTO.class);
-        } else {
-            fail("The returned entity is not of type EventBookingDTO");
+        assertEquals(DetailedEventBookingDTO.class.getCanonicalName(), getEventBookingsByIdResponse.getEntity().getClass().getCanonicalName());
+        assertNotNull(((DetailedEventBookingDTO) getEventBookingsByIdResponse.getEntity()).getUserBooked());
+        assertEquals(UserSummaryWithEmailAddressDTO.class.getCanonicalName(), ((EventBookingDTO) getEventBookingsByIdResponse.getEntity()).getUserBooked().getClass().getCanonicalName());
+
+        // --- Delete the booking created above otherwise the other tests may be affected.
+        HttpServletRequest cancelBookingRequest = createRequestWithCookies(new Cookie[] { studentLogin.cookie });
+        replay(cancelBookingRequest);
+        Response cancelBookingResponse = eventsFacade.cancelBooking(cancelBookingRequest, "_regular_test_event");
+        assertEquals(Response.Status.NO_CONTENT.getStatusCode(), cancelBookingResponse.getStatus());
+
+        // --- Tear down
+        // This should not be necessary, but we don't actually remove cancelled bookings from the database.
```

Comment:
There are a couple of comments about things which should be unnecessary but are here nonetheless, which is a bit confusing - it would be good to remove either the unnecessary thing or the comment.

Reply:
The unnecessary thing is actually necessary. It would be unnecessary if we actually removed the record from the database, but we don't, and this leaves state hanging around confusing other test cases. I might remove the comment, but leave a remark so we are aware of why the tear down phase is there.